src/hash_to_curve/map_g2.rs

```diff
@@ -0,0 +1,900 @@
+            0x6258_f7a6_232c_ab9b,
+            0x0b98_5811_cce1_4db5,
+        ]),
+    },
+    Fp2 {
+        c0: Fp::from_raw_unchecked([
+            0x8671_6401_f7f7_377b,
+            0xa31d_b74b_f3d0_3101,
+            0x1423_2543_c645_9a3c,
+            0x0a29_ccf6_8744_8752,
+            0xe8c2_b010_201f_013c,
+            0x0e68_b9d8_6c9e_98e4,
+        ]),
+        c1: Fp::from_raw_unchecked([
+            0x05e5_1466_8ac7_36d2,
+            0x9089_b4d6_b84f_3ea5,
+            0x603c_384c_224a_8b32,
+            0xf325_7909_536a_fea6,
+            0x5c5c_dbab_ae65_6d81,
+            0x075b_fa08_63c9_87e9,
+        ]),
+    },
+    Fp2 {
+        c0: Fp::from_raw_unchecked([
+            0x718f_dad2_4ee1_d90f,
+            0xa58c_025b_ed82_76af,
+            0x0c3a_1023_0ab7_976f,
+            0xf0c5_4df5_c8f2_75e1,
+            0x4ec2_478c_28ba_f465,
+            0x1129_373a_90c5_08e6,
+        ]),
+        c1: Fp::from_raw_unchecked([
+            0x019a_f5f9_80a3_680c,
+            0x4ed7_da0e_6606_3afa,
+            0x6003_5472_3b5d_9972,
+            0x8b2f_958b_20d0_9d72,
+            0x0474_938f_02d4_61db,
+            0x0dcf_8b9e_0684_ab1c,
+        ]),
+    },
+    Fp2 {
+        c0: Fp::from_raw_unchecked([
+            0xb864_0a06_7f5c_429f,
+            0xcfd4_25f0_4b4d_c505,
+            0x072d_7e2e_bb53_5cb1,
+            0xd947_b5f9_d2b4_754d,
+            0x46a7_1427_4077_4afb,
+            0x0c31_864c_32fb_3b7e,
+        ]),
+        c1: Fp::from_raw_unchecked([
+            0x718f_dad2_4ee1_d90f,
+            0xa58c_025b_ed82_76af,
+            0x0c3a_1023_0ab7_976f,
+            0xf0c5_4df5_c8f2_75e1,
+            0x4ec2_478c_28ba_f465,
+            0x1129_373a_90c5_08e6,
+        ]),
+    },
+];
+
+const SSWU_RV1: Fp2 = Fp2 {
+    c0: Fp::from_raw_unchecked([
+        0x7bcf_a7a2_5aa3_0fda,
+        0xdc17_dec1_2a92_7e7c,
+        0x2f08_8dd8_6b4e_bef1,
+        0xd1ca_2087_da74_d4a7,
+        0x2da2_5966_96ce_bc1d,
+        0x0e2b_7eed_bbfd_87d2,
+    ]),
+    c1: Fp::from_raw_unchecked([
+        0x7bcf_a7a2_5aa3_0fda,
+        0xdc17_dec1_2a92_7e7c,
+        0x2f08_8dd8_6b4e_bef1,
+        0xd1ca_2087_da74_d4a7,
+        0x2da2_5966_96ce_bc1d,
+        0x0e2b_7eed_bbfd_87d2,
+    ]),
+};
+
+impl HashToField for Fp2 {
+    // ceil(log2(p)) = 381, m = 2, k = 128.
+    type InputLength = U128;
+
+    fn from_okm(okm: &GenericArray<u8, U128>) -> Fp2 {
+        let c0 = <Fp as HashToField>::from_okm(GenericArray::<u8, U64>::from_slice(&okm[..64]));
+        let c1 = <Fp as HashToField>::from_okm(GenericArray::<u8, U64>::from_slice(&okm[64..]));
+        Fp2 { c0, c1 }
+    }
+}
+
+impl Sgn0 for Fp2 {
+    fn sgn0(&self) -> Choice {
+        let sign_0 = self.c0.sgn0();
+        let zero_0 = self.c0.is_zero();
+        let sign_1 = self.c1.sgn0();
+        sign_0 | (zero_0 & sign_1)
+    }
+}
+
+/// Maps from an [`Fp2]` element to a point on iso-G2.
+fn map_to_curve_simple_ssw(u: &Fp2) -> G2Projective {
```

Comment:
```suggestion
fn map_to_curve_simple_swu(u: &Fp2) -> G2Projective {
```
(and similarly below)